cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/PreferencesHttpHandler.java

```diff
@@ -82,7 +88,12 @@ public void deleteInstancePrefs(HttpRequest request, HttpResponder responder) th
   public void setInstancePrefs(FullHttpRequest request, HttpResponder responder) throws Exception {
     try {
       Map<String, String> propMap = decodeArguments(request);
-      preferencesStore.setProperties(propMap);
+      if (propMap.containsKey(SystemArguments.PROFILE_NAME)) {
```

Comment:
this is confusing logic. Seems like we should always just set the preferences and leave the validation to the service. Also confused why there are two ways to write preferences. I think the PreferencesStore should be renamed to PreferencesService, and that it should use a ConfigDataset and ProfileDataset inside of it to validate properties before setting them.

Reply:
similar comment in the rest of this handler. In generals, handlers should not be doing much more than validating user input and calling services.

Reply:
Removed this logic and add the logic in PreferenceStore